Formula/telnetd.rb

```diff
@@ -0,0 +1,63 @@
+class Telnetd < Formula
+  desc "TELNET server (built from macOS Sierra sources)"
+  homepage "https://opensource.apple.com/"
+  url "https://opensource.apple.com/tarballs/remote_cmds/remote_cmds-54.50.1.tar.gz"
+  sha256 "156ddec946c81af1cbbad5cc6e601135245f7300d134a239cda45ff5efd75930"
+
+  depends_on :xcode => :build
+
+  resource "libtelnet" do
+    url "https://opensource.apple.com/tarballs/libtelnet/libtelnet-13.tar.gz"
+    sha256 "e7d203083c2d9fa363da4cc4b7377d4a18f8a6f569b9bcf58f97255941a2ebd1"
+  end
+
+  def install
+    resource("libtelnet").stage do
+      xcodebuild "SYMROOT=build"
+
+      libtelnet_dst = buildpath/"telnetd.tproj/build/Products"
+      libtelnet_dst.install "build/Release/libtelnet.a"
+      libtelnet_dst.install "build/Release/usr/local/include/libtelnet/"
+    end
+
+    system "make",
+           "-C", "telnetd.tproj",
+           "OBJROOT=build/Intermediates",
+           "SYMROOT=build/Products",
+           "DSTROOT=build/Archive",
+           "CFLAGS=$(CC_Flags) -isystembuild/Products/",
+           "LDFLAGS=$(LD_Flags) -Lbuild/Products/"
+
+    bin.install "telnetd.tproj/build/Products/telnetd"
+    man8.install "telnetd.tproj/telnetd.8"
+  end
+
+  def caveats
+    <<-EOS.undent
+    If you need this program to run automatically at login or whatever, there is
+    a plist file here, at `telnetd.tproj/telnet.plist':
+    https://opensource.apple.com/tarballs/remote_cmds/remote_cmds-54.50.1.tar.gz
+
+    You can feed that into launchd by sticking it in a LaunchAgents or
+    LaunchDaemons folder. launchd may also need super-user privileges to run
+    this daemon, so you may wish to install it to /Library/LaunchAgents,
+    /Library/LaunchDaemons, /System/Library/LaunchAgents, or
+    /System/Library/LaunchDaemons.
+
+    Yet another option for auto-running this daemon is cron. You could use the
+    @reboot macro or similar.
+    EOS
+  end
+
+  test do
```

Comment:
Unless I'm missing something here this whole block can be squashed into:
```ruby
test do
  assert_match "usage: telnetd", shell_output("#{bin}/telnetd usage")
end
```

Reply:
@DomT4 The test fails when you do that. For some reason, it seems that `telnetd` is not outputting to the stdout. You can verify that like so:

```sh
$ telnetd usage > ~/test
usage: telnetd [-debug] [-D (options|report|exercise|netdata|ptydata)]
	 [-h] [-k] [-l] [-n]
	 [-u utmp_hostname_length] [-U] [port]
$ cat ~/test
$ 
```

Hence the business with `PTY` and so forth.

Reply:
What about if you use `2>&1` at the end of that `shell_output` line.

```ruby
assert_match "usage: telnetd", shell_output("#{bin}/telnetd usage 2>&1")
```